src/rgw/rgw_reshard.cc

```diff
@@ -182,23 +183,25 @@ class BucketReshardShard {
 
 
 class BucketReshardManager {
+  const DoutPrefixProvider *dpp;
```

Comment:
Here is a `dpp` declared in a class in rgw_reshard.cc.

Reply:
@cbodley this class's constructor has something that needs a dpp. The only place the constructor is called is another function that already takes a dpp so I think this one is good.

Reply:
if only the constructor needs it, we shouldn't store have to in a member variable